doc/ug/getting_started_build_sw.md

```diff
@@ -95,6 +95,159 @@ $ ./meson_init -t /path/to/toolchain/file
+OpenTitan targets (like it does with bazel\_embedded) or to satisfy dependencies
+(as it does with abseil). As we produce increasingly stable releases we should
+fix a number of these references to specific commits and not the default branch
+head.
+
+#### BUILD files
+
+Throughout the OpenTitan repo, BUILD files describe targets and dependencies in
+the same directory and subdirectories that lack their own BUILD files. We don’t
+currently have plans to automatically generate BUILD files, and to suit the
+request we don’t include handwritten files in autogen directories, you’ll see
+some big BUILD files that describe how to build and depend on auto generated
+files in autogen directories.
+
+### Linting Starlark
+
+The Linter for Bazel files is called “buildifier” and can be built and run with
+Bazel by running:
+
+```console
+bazel run //buildifier_fix
+```
+
+Which will run the linter on all of the WORKSPACE,BUILD and .bzl files in the
+workspace. It’s fairly strict so the best way to keep git logs and blame reports
+clean is to run it before committing, and ask for it if it looks like it has
+been forgotten within a PR.
+
+### Test Execution
+
+Bazel will execute tests when invoked with ```bazel test <label-expression>```.  The
+test rules have various tags applied to them to allow for filtering of tests
+during dependency analysis.  Our tests are typically tagged with verilator,
+cw310, or manual.  The ```--test_tag_filters``` switch can be used to filter out tests
+that you either cannot or do not want to execute.
+
+For example, if you do not have a CW310 FPGA board, you cannot execute the tests
+meant to execute on that board.  You can instruct bazel to not execute those
+tests:
+
+```console
+bazel test --test_tag_filters=-cw310 //...
+```
+
+Rather than manually adding that command line switch to every invocation of
+bazel, you can add it to your $HOME/.bazelrc file.
+
+
+$HOME/.bazelrc:
+test --test_tag_filters=-cw310
+
+### Excluding Verilator
+
+Verilator is slow to build and Verilator tests are slow to run. Many directories
+have Verilator targets that will cause them to build Verilator when you build
+all targets with the wildcard `...`. Tagging allows us to exclude verilator
+tests, but doesn't exclude building verilator for targets that depend on it. To
+exclude all targets that depend on verilator you should use a target pattern
+that can be generated with Bazel queries, and sed and then passed with xargs:
+
+```console
+bazel query "rdeps(//..., //hw:verilator)" \
+| sed -e 's/^/-/' \
+| xargs bazel test -- //...
+```
+
+The bazel query lists all the dependencies in the workspace that depend on
+verilator. These are inverted for the target pattern by prepending a dash and
+then the test is invoked by xargs on a pattern which starts with the whole
+workspace and then excludes each of the targets that depend on verilator.
+
+### Caching on disk
+
+If you’re building verilator and you’re tired of waiting for it to build and
```

Comment:
nit: I think this sentence could be shortened.

Reply:
Sorry for editorializing